export/mlab_export.py

```diff
@@ -0,0 +1,605 @@
+    self._handle.close()
+
+
+def get_mtime(last_export_filename):
+  """Returns the file mtime, or zero if last_export_filename was created.
+
+  Args:
+    last_export_filename: str, absolute path to last export time stamp file.
+  Returns:
+    int, 0 if the file was created, or the mtime of existing file.
+  Raises:
+    IOError, if last_export_filename does not exist and cannot be created.
+    OSError, if last_export_filename exists but stat info cannot be read.
+  """
+  if not os.path.exists(last_export_filename):
+    open(last_export_filename, 'w').close()
+    # Indicate that there is no timestamp, so the caller can use a default.
+    return 0
+  return int(os.stat(last_export_filename).st_mtime)
+
+
+def update_mtime(last_export_filename, mtime):
+  """Updates the atime & mtime on the export timestamp file.
+
+  Args:
+    last_export_filename: str, absolute path to last export time stamp file.
+    mtime: int, timestamp in seconds since epoch; used for file atime & mtime.
+  Raises:
+    OSError, if last_export_filename mtime cannot be updated.
+  """
+  os.utime(last_export_filename, (mtime, mtime))
+
+
+def align_timestamp(timestamp, step):
+  """Adjusts 'timestamp' down to a multiple of 'step' size.
+
+  Args:
+    timestamp: int, timestamp in seconds since the epoch.
+    step: int, interval to align.
+  Returns:
+    int, timestamp adjusted to multiple of step.
+  """
+  return timestamp - (timestamp % step)
+
+
+def default_start_time(options, ts_previous):
+  """Calculates a default start timestamp.
+
+  Args:
+    options: flags.FlagValues, the runtime options. These values are read:
+        options.length, options.step, options.update, options.ts_offset.
+    ts_previous: int, timestamp in seconds since epoch of last
+        successful export. On first run, this value should be zero.
+  Returns:
+    int, timestamp in seconds since the epoch.
+  """
+  if ts_previous:
+    # Typical: start from the end time of previous runs.
+    return align_timestamp(ts_previous, options.step)
+
+  ts_current = int(time.time())
+
+  # Since ts_previous is not set, this is a "first run" scenario.
+  if not options.update:
+    # Unlikely: first run by a user. Start as close to 'now' as possible.
+    start = ts_current - options.length - options.ts_offset
+  else:
+    # Likely: first, automated export. Start at previous 'length' aligned time.
+    start = align_timestamp(ts_current, options.length) - options.length
+
+  # Align end ts to a multiple of step size (just in case).
+  return align_timestamp(start, options.step)
+
+
+def assert_start_and_end_times(options):
+  """Performs a sanity check on start and end timestamps.
+
+  This method asserts that both ts_end is less than ts_start and that the
+  difference between them is greater than options.length.
+
+  Args:
+    options: flags.FlagValues, the runtime options. These values are read:
+        options.length, options.ts_start, options.ts_end.
+  Returns:
+    True on success.
```

Comment:
I missed this comment from the last round.

My goal here was to facilitate testing. I imagined the successful test case for `assert_start_and_end_times` needed to verify something. I see how the docstring could cause some confusion.

Could this be a special case for testing, where since the method under test is an "assert" operation, no other checks for success are necessary? It felt incomplete to have the test call the method under test and nothing else.

How would you handle this test case?